packages/world/src/systems/CraftFuelSystem.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.24;
+
+import { System } from "@latticexyz/world/src/System.sol";
+
+import { ObjectType } from "../codegen/tables/ObjectType.sol";
+
+import { InventorySlot, InventorySlotData } from "../codegen/tables/InventorySlot.sol";
+import { ObjectTypeMetadata } from "../codegen/tables/ObjectTypeMetadata.sol";
+
+import { CRAFT_ENERGY_COST } from "../Constants.sol";
+import { EntityId } from "../EntityId.sol";
+import { ObjectTypeId } from "../ObjectTypeId.sol";
+import { ObjectTypeLib } from "../ObjectTypeLib.sol";
+import { ObjectTypes } from "../ObjectTypes.sol";
+
+import { Vec3 } from "../Vec3.sol";
+import { transferEnergyToPool } from "../utils/EnergyUtils.sol";
+import { InventoryUtils, SlotAmount } from "../utils/InventoryUtils.sol";
+import { CraftNotification, notify } from "../utils/NotifUtils.sol";
+
+contract CraftFuelSystem is System {
```

Comment:
Tried putting this in `CraftSystem` and combining `craftFuel` and `fuelMachine` into one system but in both cases contract too big